src/dappsExplorer/DAppsExplorerScreenFilter.tsx

```diff
@@ -0,0 +1,319 @@
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import {
+  dappFavoritesEnabledSelector,
+  dappsCategoriesAlphabeticalSelector,
+  dappsListErrorSelector,
+  dappsListLoadingSelector,
+  dappsListSelector,
+  dappsMinimalDisclaimerEnabledSelector,
+  favoriteDappIdsSelector,
+} from 'src/dapps/selectors'
+import { fetchDappsList } from 'src/dapps/slice'
+import { DappFilter, DappSection, DappV1, DappV2 } from 'src/dapps/types'
+import DappCard from 'src/dappsExplorer/DappCard'
+import DappFilterChip from 'src/dappsExplorer/DappFilterChip'
+import FavoriteDappsSection from 'src/dappsExplorer/FavoriteDappsSection'
+import { NoResults } from 'src/dappsExplorer/NoResults'
+import useDappFavoritedToast from 'src/dappsExplorer/useDappFavoritedToast'
+import useDappInfoBottomSheet from 'src/dappsExplorer/useDappInfoBottomSheet'
+import useOpenDapp from 'src/dappsExplorer/useOpenDapp'
+import { currentLanguageSelector } from 'src/i18n/selectors'
+import Help from 'src/icons/Help'
+import DrawerTopBar from 'src/navigator/DrawerTopBar'
+import { TopBarIconButton } from 'src/navigator/TopBarButton'
+import colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+
+const AnimatedSectionList =
+  Animated.createAnimatedComponent<SectionListProps<DappV2, SectionData>>(SectionList)
+
+interface SectionData {
+  data: DappV2[]
+  category: string
+}
+
+export function DAppsExplorerScreenFilter() {
+  const { t } = useTranslation()
+  const insets = useSafeAreaInsets()
+
+  const sectionListRef = useRef<SectionList>(null)
+  const scrollPosition = useRef(new Animated.Value(0)).current
+  const horizontalScrollView = useRef<ScrollView>(null)
+
+  const onScroll = Animated.event([{ nativeEvent: { contentOffset: { y: scrollPosition } } }])
+  const dispatch = useDispatch()
+  const loading = useSelector(dappsListLoadingSelector)
+  const error = useSelector(dappsListErrorSelector)
+  const categories = useSelector(dappsCategoriesAlphabeticalSelector)
+  const dappFavoritesEnabled = useSelector(dappFavoritesEnabledSelector)
+  const dappsMinimalDisclaimerEnabled = useSelector(dappsMinimalDisclaimerEnabledSelector)
+  const dappList = useSelector(dappsListSelector)
+  const language = useSelector(currentLanguageSelector)
+  const favoriteDappsById = useSelector(favoriteDappIdsSelector)
+  const [selectedFilter, setSelectedFilter] = useState({
+    id: 'all',
+    name: t('dappsScreen.allDapps'),
+  })
+
+  const { onSelectDapp, ConfirmOpenDappBottomSheet } = useOpenDapp()
+  const { onFavoriteDapp, DappFavoritedToast } = useDappFavoritedToast(sectionListRef)
+  const { openSheet, DappInfoBottomSheet } = useDappInfoBottomSheet()
+
+  useEffect(() => {
+    dispatch(fetchDappsList())
+    ValoraAnalytics.track(DappExplorerEvents.dapp_screen_open)
+  }, [])
+
+  const removeFilter = () => {
+    ValoraAnalytics.track(DappExplorerEvents.dapp_filter, { id: selectedFilter.id, remove: true })
```

Comment:
could we make the set / remove filter implementation consistent? i don't mind which solution to take, but it would be nice to either declare the set / remove filter functions in this file and pass it down OR pass the `setSelectedFilter` down directly. i.e. manage the same analytics event in the same file.

right now, the remove filter is declared here and passed to the favourites and no results components. meanwhile the filter chip component takes the `setSelectedFilter` directly and handles the analytics. i know it probably made sense to keep the `removeFilter` here because of the scroll ref, but i think it'd be nice to have the chip filter component also just receive a callback that it can execute without worrying about any extra logic. what do you think?